odxtools/parameterinfo.py

```diff
@@ -1,105 +1,187 @@
+                    textwrap.indent(parameter_info(mux_case.structure.parameters, True), "    "))
+                of.write(f"   }})\n")
             continue
 
-        result += f"{param.short_name}"
+        of.write(f"{q}{param.short_name}{q}")
 
         if dop is None:
-            result += ": <no DOP>\n"
+            of.write(": <no DOP>\n")
             continue
         elif not isinstance(dop, DataObjectProperty):
-            result += ": <unhandled DOP>\n"
+            of.write(f": <unhandled DOP '{type(dop).__name__}'>\n")
             continue
 
         if (cm := dop.compu_method) is None:
-            result += ": <no compu method>\n"
+            of.write(": <no compu method>\n")
             continue
 
         if isinstance(cm, TexttableCompuMethod):
-            result += f": enum; choices:\n"
+            of.write(f": enum; choices:\n")
             for scale in cm.internal_to_phys:
-                result += f"  '{str(scale.compu_const)}'\n"
+                val_str = ""
+                if scale.lower_limit is not None:
+                    val_str = f"({repr(scale.lower_limit.value)})"
+                of.write(f"  {repr(scale.compu_const)}{val_str}\n")
 
         elif isinstance(cm, IdenticalCompuMethod):
             bdt = dop.physical_type.base_data_type
             if bdt in (DataType.A_UTF8STRING, DataType.A_UNICODE2STRING, DataType.A_ASCIISTRING):
-                result += f": str"
-            elif bdt in (DataType.A_BYTEFIELD,):
-                result += f": bytes"
+                of.write(f": str")
+            elif bdt == DataType.A_BYTEFIELD:
+                of.write(f": bytes")
             elif bdt.name.startswith("A_FLOAT"):
-                result += f": float"
+                of.write(f": float")
             elif bdt.name.startswith("A_UINT"):
-                result += f": uint"
+                of.write(f": uint")
             elif bdt.name.startswith("A_INT"):
-                result += f": int"
+                of.write(f": int")
             else:
-                result += f": <unknown type>"
+                of.write(f": <unknown type {{ bdt.name }}>")
 
-            if (bl := dop.get_static_bit_length()) is not None:
-                result += f"{bl}"
-
-            result += "\n"
+            of.write("\n")
 
         elif isinstance(cm, LinearCompuMethod):
-            result += f": float\n"
+            bdt = dop.physical_type.base_data_type
+            if bdt in (DataType.A_UTF8STRING, DataType.A_UNICODE2STRING, DataType.A_ASCIISTRING):
+                of.write(f": str")
+            elif bdt in (DataType.A_BYTEFIELD,):
+                of.write(f": bytes")
+            elif bdt.name.startswith("A_FLOAT"):
+                of.write(f": float")
+            elif bdt.name.startswith("A_UINT"):
+                of.write(f": uint")
+            elif bdt.name.startswith("A_INT"):
+                of.write(f": int")
+            else:
+                of.write(f": <unknown type>")
+
```

Comment:
I think it would be better to move this logic to the `__str__` of the physical datatype class